cmd/kubeadm/app/apis/kubeadm/timeoututils.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeadm
+
+import (
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetDefaultTimeouts sets an internal Timeouts struct to its default values.
+func SetDefaultTimeouts(t **Timeouts) {
+	*t = &Timeouts{
+		ControlPlaneComponentHealthCheck: &metav1.Duration{Duration: 4 * time.Minute},
+		KubeletHealthCheck:               &metav1.Duration{Duration: 4 * time.Minute},
+		KubernetesAPICall:                &metav1.Duration{Duration: 1 * time.Minute},
+		EtcdAPICall:                      &metav1.Duration{Duration: 2 * time.Minute},
+		TLSBootstrap:                     &metav1.Duration{Duration: 5 * time.Minute},
+		Discovery:                        &metav1.Duration{Duration: 5 * time.Minute},
```

Comment:
Whether it is better to use defined constants instead of literals?
for example:
```go
   ...
   Discovery: &metav1.Duration{Duration: constants.DefaultDiscoveryTimeout},
   ...
```

Reply:
added as part of commit:
`kubeadm: start using the Timeouts struct values`

this is where all constants were refactored and defaults started to be used in the public v1beta4.

Reply:
to clarify: i had to add this defaulter util function because the converter for v1beta3 and the fuzzer needed it.
we might be able to drop it once v1beta3 is removed.

Reply:
/lgtm